examples/turbulent_advection-diffusion.jl

```diff
@@ -0,0 +1,169 @@
+ H = [0.2, 0.8]          # the rest depths of each layer
+ ρ = [4.0, 5.0]          # the density of each layer
+
+ U = zeros(nlayers) # the imposed mean zonal flow in each layer
+ U[1] = 1.0
+ U[2] = 0.0
+
+# ### `MultiLayerQG.Problem` setup, shortcuts and initial conditions
+MQGprob = MultiLayerQG.Problem(nlayers, dev;
+                        nx=n, Lx=L, f₀=f₀, g=g, H=H, ρ=ρ, U=U, μ=μ, β=β,
+                        dt=dt, stepper=stepper, aliased_fraction=0)
+grid = MQGprob.grid
+x, y = grid.x, grid.y
+
+# Initial conditions                        
+seed!(1234) # reset of the random number generator for reproducibility
+q₀  = 1e-2 * ArrayType(dev)(randn((grid.nx, grid.ny, nlayers)))
+q₀h = MQGprob.timestepper.filter .* rfft(q₀, (1, 2)) # apply rfft  only in dims=1, 2
+q₀  = irfft(q₀h, grid.nx, (1, 2))                 # apply irfft only in dims=1, 2
+
+MultiLayerQG.set_q!(MQGprob, q₀)
+nothing 
+
+# ## Tracer advection-diffusion setup
+# Now that we have a `MultiLayerQG.Problem` setup to generate our turbulent flow, we
+# setup an advection-diffusion simulation. This is done by passing the `MultiLayerQG.Problem`
+# as an argument to `TracerAdvectionDiffusion.Problem` which sets up an advection-diffusion problem
+# with same parameters where applicable. We also need to pass a value for the constant diffusivity `κ`,
+# the `stepper` used to step the problem forward and when we want the tracer released into the flow.
+# We will let the flow run until it reaches a statistical equilibrium and then advect-diffuse the tracer.
+
+κ = 0.002
+nsteps = 4000               # total number of time-steps
+nsubs = 1                   # number of steps the simulation takes at each iteration 
+tracer_release = dt * 8000  # run flow for some time before releasing tracer
+
+ADprob = TracerAdvectionDiffusion.Problem(dev, MQGprob; κ, stepper, tracer_release)
+nothing
+
+# ## Initial condition for concentration in both layers
+# We have a two layer system so we will advect-diffuse the tracer in both layers.
+# To do this we set the initial condition for tracer concetration as a Gaussian centred at the origin.
+# Then we create some shortcuts for the `TracerAdvectionDiffusion.Problem`.
+gaussian(x, y, σ) = exp(-(x^2 + y^2) / (2σ^2))
+
+amplitude, spread = 10, 0.15
+c₀ = [amplitude * gaussian(x[i], y[j], spread) for i=1:grid.nx, j=1:grid.ny]
+
+TracerAdvectionDiffusion.set_c!(ADprob, c₀, nlayers)
+
+# Shortcuts for advection-diffusion problem
+sol, clock, vars, params, grid = ADprob.sol, ADprob.clock, ADprob.vars, ADprob.params, ADprob.grid
+x, y = grid.x, grid.y
+
+# ## Saving output
+# The parent package `FourierFlows.jl` provides a way to save information from a simulation.
+# To do this we write a function `GetConcentration` and pass this to the `Output` function along 
+# with the `TracerAdvectionDiffusion.Problem` and the name of the output file.
+
+function GetConcentration(prob)
+    Concentration = @. prob.vars.c
+    return Concentration
+end
```

Comment:
```suggestion
get_concentration(prob) = prob.vars.c
```

Reply:
This version of `get_concentration` implies that `updatevars!` was called so that `prob.vars.c` contains the latest update. Perhaps it'd be safer to write something like
```Julia
function get_concentration(prob):
    ldiv!(prob.vars.c, prob.grid.rfftplan, deepcopy(sol))
    return prob.vars.c
end
```

?

Reply:
Yes that is true! I will make this change.